kore/src/Kore/Step/Simplification/Pattern.hs

```diff
@@ -88,7 +90,7 @@ simplifyPredicate
     -- ^ The condition to be evaluated.
     -> BranchT simplifier (Pattern variable)
 simplifyPredicate Conditional {term, predicate, substitution} = do
-    evaluated <- Monad.Trans.lift $ Predicate.evaluate predicate
+    evaluated <- Monad.Trans.lift $ Predicate.simplify predicate
```

Comment:
I'm not sure what you meant, but: I have to convert a `simplifier` (returned by `Predicate.simplify`) to a `BranchT simplifier` (required by the function signature), and I can't change the function's signature since `mergePredicatesAndSubstitutions` uses `BranchT simplifier`, and we're usually converting `simplifier` to `BranchT simplifier` whenever we can, not the other way around.